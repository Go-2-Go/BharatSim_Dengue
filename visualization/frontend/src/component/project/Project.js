import React from 'react';
import DashboardNavbar from './DashboardNavbar';
import SideBarLayout from '../layout/SideBarLayout';
import DashboardView from './DashboardView';

function Project() {
  const [selectedDashboard, setSelectedDashboard] = React.useState(0);

  const handleChange = (event, newValue) => {
    setSelectedDashboard(newValue);
  };

  const controllers = ['Dashboard 1'];
  const views = [{ name: 'dashboard1' }];
  return (
    <SideBarLayout
      ControllerComponent={() => (
        <DashboardNavbar
          controllers={controllers}
          value={selectedDashboard}
          handleChange={handleChange}
        />
      )}
      ViewComponent={() => <DashboardView views={views} value={selectedDashboard} />}
    />
  );
}

export default Project;
