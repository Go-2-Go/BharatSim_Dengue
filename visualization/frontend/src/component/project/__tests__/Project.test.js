import React from 'react';
import { render } from '@testing-library/react';
import Project from '../Project';

describe('Project', () => {
  it('should match snapshot', () => {
    const { container } = render(<Project />);

    expect(container).toMatchSnapshot();
  });
});
