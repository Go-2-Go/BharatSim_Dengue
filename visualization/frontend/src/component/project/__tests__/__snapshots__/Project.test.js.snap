// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match snapshot for old projects 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-38 makeStyles-layoutContainer-34"
  >
    <div
      class="MuiBox-root MuiBox-root-39 makeStyles-sideBarLayout-35"
    >
      <div
        class="MuiBox-root MuiBox-root-46"
      >
        <div
          class="MuiTabs-root MuiTabs-vertical"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              class="MuiTabs-flexContainer MuiTabs-flexContainerVertical"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root makeStyles-root-40 MuiTab-textColorInherit makeStyles-textColorInherit-44 Mui-selected makeStyles-selected-43 MuiTab-labelIcon makeStyles-labelIcon-42 MuiTab-fullWidth"
                data-testid="tab-dashboard1"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper makeStyles-wrapper-41"
                >
                  <div
                    class="MuiBox-root MuiBox-root-47"
                  >
                    <img
                      alt="dashboard-logo"
                      src="dashboard-icon.svg"
                    />
                  </div>
                  dashboard1
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root-48 PrivateTabIndicator-colorSecondary-50 MuiTabs-indicator makeStyles-indicator-45 PrivateTabIndicator-vertical-51"
              style="top: 0px; height: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-52"
    >
      <div
        class="MuiBox-root MuiBox-root-53 makeStyles-projectNameBar-33"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          project1
        </h5>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Save
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-54"
      >
        <div
          aria-labelledby="simple-tab-0"
          id="simple-tabpanel-0"
          role="tabpanel"
        >
          <div
            class="MuiBox-root MuiBox-root-59"
          >
            <div
              class="MuiBox-root MuiBox-root-60 makeStyles-configureProjectDataBar-55"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                 Configure Dashboard Data
              </h6>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                   Back to recent projects
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root MuiBox-root-61 makeStyles-dashboardDataContainer-57"
            >
              <div
                class="MuiBox-root MuiBox-root-62"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                   Manage Dashboard Dataset
                </h6>
              </div>
              <div
                class="MuiBox-root MuiBox-root-63 makeStyles-dashboardDataHeader-56"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  project1 :: dashboard1
                </h6>
              </div>
              <div
                class="MuiBox-root MuiBox-root-64 makeStyles-dashboardDataBody-58"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
                >
                  Before we can create any visualization, we ‘ll need some data.
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Use Upload dataset to add data files to your dashboard.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Project should match snapshot while creating new project 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6 makeStyles-layoutContainer-2"
  >
    <div
      class="MuiBox-root MuiBox-root-7 makeStyles-sideBarLayout-3"
    >
      <div
        class="MuiBox-root MuiBox-root-14"
      >
        <div
          class="MuiTabs-root MuiTabs-vertical"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              class="MuiTabs-flexContainer MuiTabs-flexContainerVertical"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root makeStyles-root-8 MuiTab-textColorInherit makeStyles-textColorInherit-12 Mui-selected makeStyles-selected-11 MuiTab-labelIcon makeStyles-labelIcon-10 MuiTab-fullWidth"
                data-testid="tab-dashboard1"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper makeStyles-wrapper-9"
                >
                  <div
                    class="MuiBox-root MuiBox-root-15"
                  >
                    <img
                      alt="dashboard-logo"
                      src="dashboard-icon.svg"
                    />
                  </div>
                  dashboard1
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root-16 PrivateTabIndicator-colorSecondary-18 MuiTabs-indicator makeStyles-indicator-13 PrivateTabIndicator-vertical-19"
              style="top: 0px; height: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-20"
    >
      <div
        class="MuiBox-root MuiBox-root-21 makeStyles-projectNameBar-1"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          untitled project
        </h5>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Save
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-22"
      >
        <div
          aria-labelledby="simple-tab-0"
          id="simple-tabpanel-0"
          role="tabpanel"
        >
          <div
            class="MuiBox-root MuiBox-root-27"
          >
            <div
              class="MuiBox-root MuiBox-root-28 makeStyles-configureProjectDataBar-23"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                 Configure Dashboard Data
              </h6>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                   Back to recent projects
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root MuiBox-root-29 makeStyles-dashboardDataContainer-25"
            >
              <div
                class="MuiBox-root MuiBox-root-30"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                   Manage Dashboard Dataset
                </h6>
              </div>
              <div
                class="MuiBox-root MuiBox-root-31 makeStyles-dashboardDataHeader-24"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  untitled project :: dashboard1
                </h6>
              </div>
              <div
                class="MuiBox-root MuiBox-root-32 makeStyles-dashboardDataBody-26"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
                >
                  Before we can create any visualization, we ‘ll need some data.
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  Use Upload dataset to add data files to your dashboard.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
